go/ethclient/wallet/wallet.go

```diff
@@ -3,21 +3,30 @@ package wallet
 import (
 	"crypto/ecdsa"
 	"math/big"
+	"sync/atomic"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
 type Wallet interface {
+	// Address returns the pubkey Address of the wallet
 	Address() common.Address
+	// SignTransaction returns a signed transaction
 	SignTransaction(chainID int, tx types.TxData) (*types.Transaction, error)
+
+	// SetNonce overrides the current nonce
+	SetNonce(nonce uint64)
```

Comment:
you should do a +1 inside the implementation

Reply:
I feel that might be misleading ?
I've change a bit the naming of the methods, please take a look.

Reply:
if nonce=7 on the blockchain,  you set it to the variable.
And then when you request it for the next tx it will be (7+1-1) =7

Reply:
Ah yes, I get it. We're using the `PendingNonceAt` which gives us the next nonce to use.

Unless you want to change this to the `NonceAt` which gives the account current nonce ?

Reply:
Either way, tell me what's prefered and I'll a comment explaining the approach 👍

Reply:
cool. I didn't realise that.